src/open_inwoner/configurations/admin.py

```diff
@@ -310,6 +350,104 @@ def _get_column(props):
+                    insert_index = len(fields)
+                    js_fields = [
+                        "custom_javascript_confirmed",
+                        "custom_javascript",
+                        "custom_javascript_file_info",
+                    ]
+
+                    for js_field in js_fields:
+                        if js_field in fields:
+                            insert_index = fields.index(js_field)
+                            break
+
+                    # Insert status field before JS fields when enabled
+                    fields.insert(insert_index, "custom_javascript_status")
+
+                new_options = options.copy()
+                new_options["fields"] = tuple(fields)
+                fieldsets[i] = (name, new_options)
+                break
+
+        return tuple(fieldsets)
+
+    def get_readonly_fields(self, request, obj=None):
+        readonly_fields = [
+            "extra_css_allowed",
+            "custom_javascript_status",
+        ]
+
+        if settings.ALLOW_CUSTOM_JS:
+            readonly_fields.append("custom_javascript_file_info")
+
+        return readonly_fields
+
+    @admin.display(description=_("Custom JavaScript Status"))
+    def custom_javascript_status(self, obj):
+        if settings.ALLOW_CUSTOM_JS:
+            return format_html(
+                '<span class="js-enabled">{}</span>', _("Custom JavaScript is enabled")
+            )
+        else:
+            return format_html(
+                '<span class="js-disabled">{}</span>',
+                _("Custom JavaScript is disabled. Contact your system administrator."),
+            )
+
+    @admin.display(description=_("Current upload"))
+    def custom_javascript_file_info(self, obj):
+        if obj.custom_javascript:
+            try:
+                if obj.custom_javascript.storage.exists(obj.custom_javascript.name):
+                    if settings.ALLOW_CUSTOM_JS:
+                        size_kb = obj.custom_javascript.size / 1024
+                        # Extract just the filename from the full path
+                        filename = obj.custom_javascript.name.split("/")[-1]
+                        # Filename extraction
+                        if "_" in filename and filename.endswith(".js"):
+                            base_name = filename.split("_")[0]
+                            original_filename = f"{base_name}.js"
+                        else:
+                            original_filename = filename
+
+                        return format_html(
+                            "{}: {} KB",
+                            original_filename,  # Show filename
+                            round(size_kb, 1),  # Show size
+                        )
+                    else:
+                        return _("File uploaded but feature is disabled")
+                else:
+                    return _("File missing from storage")
+            except (OSError, IOError):
```

Comment:
Let's catch all exceptions, as discussed today, we don't want a failure here to tear down the admin:


```suggestion
            except Exception:
```